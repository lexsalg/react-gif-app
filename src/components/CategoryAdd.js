import { useState } from "react";

import PropTypes from 'prop-types';

export const CategoryAdd = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('')

    const handleInputChange = (e) => {
        setInputValue(e.target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 2) {
            // setCategories([...categories, 'nuevo'])
            setCategories(state => [inputValue, ...state])
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>

            <h1>{inputValue}</h1>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}


CategoryAdd.propTypes = {
    setCategories: PropTypes.func.isRequired
}