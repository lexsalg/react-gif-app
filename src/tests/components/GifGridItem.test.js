import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem'

describe('Pruebas en <GifGridItem />', () => {

    test('should show component correctly', () => {


        const title = 'Titulo';
        const url = 'https://localhost:3000/test.png';

        const wrapper = shallow(<GifGridItem title={title} url={url} />)
        expect(wrapper).toMatchSnapshot();

    })


})