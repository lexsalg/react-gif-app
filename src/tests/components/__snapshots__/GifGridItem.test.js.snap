// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> should show component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo"
    src="https://localhost:3000/test.png"
  />
  <p>
    Titulo
  </p>
</div>
`;
